jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1904,6 +1906,33 @@ typedef struct
   uint8_t element_size; /**< element size based on [[TypedArrayName]] in Table 49 */
 } ecma_typedarray_info_t;
 
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+/**
+ * Checks whether a given typedarray is BigInt type or not.
+ *
+ * Note: this macro should be updated when new types are added
+ **/
+#define ECMA_TYPEDARRAY_IS_BIGINT_TYPE(info) \
+    ((info.id) >= ECMA_BIGINT64_ARRAY)
+
+/**
+ * Read and copy a uint64_t value from a given buffer to a value.
+ *
+ **/
+#define ECMA_TYPEDARRAY_GET_UINT64_FROM_BUFFER(src_p, digit) \
+    do \
+    { \
+      if (JERRY_LIKELY ((((uintptr_t) src) & (8 * sizeof (uint64_t) -1)) == 0)) \
```

Comment:
I thought you create a macro only for this line. Something like:
```
#define ECMA_TYPEDARRAY_POINTER_IS_ALIGNED (ptr, type) \
  ((((uintptr_t) (ptr)) & (sizeof (type) - 1)) == 0)
```
The `*8` was incorrect.
You can use this macro for all typedarray setters, e.g. uint32_t ...

But your idea can work as well. Create a getter / setter macro pair with ptr/type paramters.
Anyway, move this macro to ecma-typedarray-object.c since it will never be used outside.

Reply:
I've made a macro for the getters, but the setter methods are quite different from each other, i couldn't come up with a good solution.

Reply:
NVM i've made a macro for the setters too.